pkg/typesystem/typesystem.go

```diff
@@ -165,13 +168,31 @@ type TypeSystem struct {
 	// [objectType] => [relationName] => TTU relation.
 	ttuRelations map[string]map[string][]*openfgav1.TupleToUserset
 
-	modelID       string
-	schemaVersion string
+	modelID                 string
+	schemaVersion           string
+	authorizationModelGraph *graph.AuthorizationModelGraph
+	authorizationModelNodes map[string]*graph.AuthorizationModelNode
```

Comment:
Dang it - when I wrote the code to build the graph I used a helper map for exactly this: be able to find in o(1) a node given it's label

https://github.com/openfga/language/blob/f3eb5182c6681154ebc23cc3d9a81307cb32cfcd/pkg/go/graph/graph_builder.go#L14-L18

I think we should expose this functionality as a method on the graph itself and not have two distinct fields here storing that information. I can do that for you 😄